R/utils-ui.R

```diff
@@ -196,36 +196,31 @@ message <- function(...) {
     instead of {.fun message}")
 }
 
-warn_for_verbose <- function(verbose = TRUE, env = caller_env()) {
+warn_for_verbose <- function(verbose = TRUE,
+                             env = caller_env(),
+                             user_env = caller_env(2)) {
   # this is about whether `verbose` was present in the **user's** call to the
   # calling function
-  # don't worry about the `verbose = TRUE` default here
+  # don't worry about the `verbose = TRUE` default in warn_for_verbose()
   if (!lifecycle::is_present(verbose) || isTRUE(verbose)) {
     return(invisible())
   }
 
-  called_from <- sys.parent(1)
-  if (called_from == 0) {
-    # called from global env, presumably in a test or during development
-    caller <- "some_googledrive_function"
-  } else {
-    called_as <- sys.call(called_from)
-    if (is.call(called_as) && is.symbol(called_as[[1]])) {
-      caller <- as.character(called_as[[1]])
-    } else {
-      caller <- "some_googledrive_function"
-    }
-  }
```

Comment:
The aforementioned "DIY moves" in which I determined the name of the calling function. In this update, I am letting go of this in favour of a simpler function. I'll probably remove `verbose` and this helper entirely in the next release of googledrive anyway.